clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2209,17 +2218,67 @@ namespace {
         DumpExpression(llvm::outs(), E);
     }
 
-    void VisitUnaryOperator(UnaryOperator *E, CheckedScopeSpecifier CSS) {
-      if (E->getOpcode() == UO_AddrOf)
-        S.CheckAddressTakenMembers(E);
+    BoundsExpr *CheckUnaryOperator(UnaryOperator *E, CheckedScopeSpecifier CSS, SideEffects SE) {
```

Comment:
I think it would be helpful to add a comment of the form: "If e is an rvalue, CheckUnaryOperator returns the bounds for the value produced by e.  If e is an lvalue expression, it returns BoundsUnknown."